tests/unit/amo/components/TestSearchContextCard.js

```diff
@@ -0,0 +1,158 @@
+import { shallow } from 'enzyme';
+import React from 'react';
+
+import {
+  SearchContextCardBase,
+  mapStateToProps,
+} from 'amo/components/SearchContextCard';
+import { searchLoad, searchStart } from 'core/actions/search';
+import { dispatchClientMetadata, fakeAddon } from 'tests/unit/amo/helpers';
+import { getFakeI18nInst } from 'tests/unit/helpers';
+
+
+describe('SearchContextCard', () => {
+  function render(props = {}) {
+    let { store } = props;
+
+    if (!store) {
+      store = dispatchClientMetadata().store;
+      store.dispatch(searchStart({ filters: { query: 'test' } }));
+    }
+
+    return shallow(
+      <SearchContextCardBase
+        {...mapStateToProps(store.getState())}
```

Comment:
This is interesting. I think Enzyme gives us an opportunity to stop using the base component (which was always a hack).

However, [HOC rendering](https://github.com/airbnb/enzyme/issues/539) isn't supported very well yet.

Until it is, we could do it like this (I tested this locally). It's a bit hacky, what do you think? By rendering the real component chain I think we will get better coverage.

```diff
diff --git a/tests/unit/amo/components/TestSearchContextCard.js b/tests/unit/amo/components/TestSearchContextCard.js
index 6504093e..5e8861f4 100644
--- a/tests/unit/amo/components/TestSearchContextCard.js
+++ b/tests/unit/amo/components/TestSearchContextCard.js
@@ -1,10 +1,7 @@
 import { shallow } from 'enzyme';
 import React from 'react';
 
-import {
-  SearchContextCardBase,
-  mapStateToProps,
-} from 'amo/components/SearchContextCard';
+import SearchContextCard from 'amo/components/SearchContextCard';
 import { searchLoad, searchStart } from 'core/actions/search';
 import { dispatchClientMetadata, fakeAddon } from 'tests/unit/amo/helpers';
 import { getFakeI18nInst } from 'tests/unit/helpers';
@@ -19,13 +16,16 @@ describe('SearchContextCard', () => {
       store.dispatch(searchStart({ filters: { query: 'test' } }));
     }
 
-    return shallow(
-      <SearchContextCardBase
-        {...mapStateToProps(store.getState())}
+    const options = { context: { store } };
+
+    const root = shallow(
+      <SearchContextCard
         i18n={getFakeI18nInst()}
         {...props}
-      />
-    );
+      />, options
+    )
+
+    return root.first().shallow(options).first().shallow(options);
   }
 
   it('should render a card', () => {
```

If we add a new HOC to the chain, the test code will need an update but only in one place.